### PR TITLE
Rahul/ifl 1653 update gettransaction response to use the rpc transaction

### PR DESCRIPTION
**Pull request by patnir**
## Summary

GetTransaction now uses RpcTransaction. Depends on #4299 

## Documentation

Does this change require any updates to the Iron Fish Docs (ex. [the RPC API
Reference](https://ironfish.network/docs/onboarding/rpc/chain))? If yes, link a
related documentation pull request for the website.

```
[ ] Yes
```

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```

### CODE REVIEW

File: ironfish/src/rpc/routes/chain/getTransaction.ts [hunk 1]
```diff
@@ -112,8 +94,10 @@ routes.register<typeof GetTransactionRequestSchema, GetTransactionResponse>(
     const { transaction, initialNoteIndex } = foundTransaction
 
     const rawTransaction: GetTransactionResponse = {
-      fee: transaction.fee().toString(),
+      fee: Number(transaction.fee()),
```
Comment (hughy) [inline ironfish/src/rpc/routes/chain/getTransaction.ts:97]: The fee should probably be a string representation of a BigInt because fee is a BigInt rather than a Number

We'll have to look at what the impact of changing fee to a string in `RpcTransaction` is

Comment (patnir) [reply ironfish/src/rpc/routes/chain/getTransaction.ts:97]: I can take a look, added a task here: https://linear.app/if-labs/issue/IFL-1676/investigate-changing-fee-to-string-from-number

Comment (danield9tqh) [reply ironfish/src/rpc/routes/chain/getTransaction.ts:97]: I remember we standardized as a number in this PR https://github.com/iron-fish/ironfish/pull/4217#issuecomment-1690686845. String is def more precise though. I think I said to go with Number to avoid backwards incompatible change on a more commonly used endpoint (can't remember which).

TLDR though I think either is fine for now whichever is more backwards compatible on more endpoints imo. We'd need a fee that's like ¼ of the terminal IRON supply to overflow this

Comment (hughy) [reply ironfish/src/rpc/routes/chain/getTransaction.ts:97]: Good point, a fee like that would be a dream for a miner, but not likely to ever happen

**Review by danield9tqh — APPROVED** [review]
